### ISSUE TITLE
I got error Cannot modify dependent keys for `cellValue` when I went to other route and back

### ISSUE THREAD

**Issue by bedrosenator** [Addepar/ember-table#638]
I get data from the backend as ember data model and cast it to array for table. It works well on table's page. But when I try to go to some other page I see the error Assertion Failed: Cannot modify dependent keys for `cellValue` on `<(unknown):ember940>` after it has been destroyed

it happens here (in ember-table/-private/utils/computed)
``` javascript 
  _contentDidChange: Ember.observer('_content', function () {
      if (!this._isUpdating) {
        this._context.notifyPropertyChange(this._key);
      }
    }),
```
after this method here crash happens, because this.isDestroyed is true (ember-runtime/lib/mixins/observable)

``` javascript
  notifyPropertyChange: function (keyName) {
      (0, _emberMetal.notifyPropertyChange)(this, keyName);
      return this;
    },
```

**Comment by Frozenfire92** [comment]
I'm also running into this issue. I can confirm that the code @bedrosenator highlighted is whats causing the problem (added logs before/after).

I tried blocking the call if `this.isDestroyed` or `this.isDestroying` is true to no avail

For some context:

My app has an `x-editor` component which contains a tabbed interface showing `source-table` (ember-table inside) component. 

```
{{#tabbed-interface tabs=tabs}}
  {{#if activeSource}}
    {{source-table source=activeSource}}
  {{/if}}
{{/tabbed-interface}}
```

Switching between tabs (activeSources) doesn't trigger the error
Leaving x-editor and coming back with a different context doesn't trigger the error
Leaving x-editor and coming back to the same context always triggers the error

note I'm not actually doing route transitions, leaving/entering x-editor happens from within a parent "display" component

**Comment by ijlee2** [comment]
Hello. I don't suppose there has been progress to solving this issue? I ran into the problem today (using Ember Data and changing routes, as also seen by @bedrosenator) and am hoping for a fix soon.

**Comment by Frozenfire92** [comment]
There is a fix in #648 but I suspect it will need #656 fixed first before it can merge

**Comment by ijlee2** [comment]
@Frozenfire92 Cool, thank you!
